libethcore/SealEngine.cpp

```diff
@@ -29,6 +29,17 @@ void NoProof::init()
     ETH_REGISTER_SEAL_ENGINE(NoProof);
 }
 
+void NoProof::generateSeal(BlockHeader const& _bi)
+{
+    BlockHeader header(_bi);
+    header.setSeal(1, h64{0});   // NonceField
```

Comment:
Maybe move `enum { MixHashField = 0, NonceField = 1 };` to `SealEngineBase` and use it here

Reply:
this is a good question. what if with sealEngine = Ethash2  there will be 3 or 1 fields for signature?

Reply:
We'll see then, let's not try to guess the future